### PR TITLE
Added some basic tests

### PR DESCRIPTION
**Pull request by PietroPasotti**
It was brought to my attention (by Michele Mancioppi @ Canonical) that this charm is in need of some testing love. I am onboarding with Canonical and this was a useful exercise to learn how to use `pytest_operator` and the `ops.testing.Harness`.
I added some simple unit/integration tests, and the standard charm linting tests. 

**How was the code tested?**
`tox -e ALL`

Final checklist
- [ x ] I am the author of these changes, or I have the rights to submit them.
- [ x ] I added the relevant changes to the README and/or documentation.
- [ x ] I self reviewed my own code.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -10,11 +10,21 @@ setenv =
...
+    -r{toxinidir}/tests/unit/requirements.txt
+
+[testenv:integration]
+commands = 
+    pytest {toxinidir}/tests/integration
+deps = 
+    -r{toxinidir}/tests/integration/requirements.txt
 
 [testenv:lint]
 commands = flake8 src/
```
Comment (heitorPB) [inline tox.ini:30]:
```suggestion
commands = flake8 srctests/
```

Do you mind also linting the code?

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class COSProxyCharmTest(unittest.TestCase):
+    def setUp(self):
+        self.harness = Harness(charm.NodeExporterCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_prometheus_relation(self):
+        prometheus = self.harness.charm.prometheus
+        rel_name = prometheus._relation_name
+        self.assertEqual(prometheus._relation_name, rel_name)
```
Comment (heitorPB) [inline tests/unit/test_charm.py:20]: I don't get this. What is the purpose of checking if a value is equal to itself? `rel_name` is `prometheus._relation_name`, or am I missing something here?

**Comment by heitorPB** [comment]
Thank you @PietroPasotti! This is awesome! It was exactly what this charm needed!

I left some comments and questions here for you here :)

File: tests/integration/test_charm.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    await ops_test.model.deploy('prometheus2')
+    await ops_test.model.add_relation(
+        'prometheus2:target',
+        'telegraf:prometheus-client'
+    )
+    await ops_test.model.add_relation(
+        'prometheus2:scrape',
+        'prometheus-node-exporter:prometheus'
+    )
+    await ops_test.model.wait_for_idle(status='active')
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply tests/integration/test_charm.py:39]: I did try and find a programmatic way to do that but I ran out of time. Agree that that is a logical next step, will do it if I have time next week

File: tests/unit/test_charm.py (hunk 2 above)
Comment (PietroPasotti) [reply tests/unit/test_charm.py:25]: that was more of a sanity check, obtaining the relation name in two different ways and asserting they're equal. Didn't sound trivial to me, but if you say this is guaranteed to happen, I can remove it...

Comment (heitorPB) [reply tests/unit/test_charm.py:25]: Ah, go it!

Maybe check if the relation name from Prometheus charm matches the relation from node-exporter charm instead?

Comment (PietroPasotti) [reply tests/unit/test_charm.py]: I think the test is much better now; let me know what you think

**Comment by heitorPB** [comment]
Thank you for this PR!

**Pull request merged**